client/cart/blocks/index.js

```diff
@@ -2,11 +2,26 @@
  * Internal dependencies
  */
 import { renderBNPLCartMessaging } from './product-details';
+import { getUPEConfig } from 'wcpay/utils/checkout';
 
 const { registerPlugin } = window.wp.plugins;
 
-// Register BNPL site messaging on the cart block.
-registerPlugin( 'bnpl-site-messaging', {
-	render: renderBNPLCartMessaging,
-	scope: 'woocommerce-checkout',
-} );
+const paymentMethods = getUPEConfig( 'paymentMethodsConfig' );
+
+const BNPL_PAYMENT_METHODS = {
+	AFFIRM: 'affirm',
+	AFTERPAY: 'afterpay_clearpay',
+	KLARNA: 'klarna',
+};
+
+const bnplPaymentMethods = Object.values(
+	BNPL_PAYMENT_METHODS
+).filter( ( method ) => Object.keys( paymentMethods ).includes( method ) );
```

Comment:
Really optional as the current solution works well: I'm curious if the following would optimize the code a bit: 
```suggestion
const bnplPaymentMethods = Object.values(BNPL_PAYMENT_METHODS).filter(
    method => method in paymentMethods
);
```

Reply:
Thanks for the suggestion! Much better 👍